lib/components/adapter.js

```diff
@@ -0,0 +1,124 @@
+/**
+ * @private
+ * @class Adapter
+ * @description Making requests to Shopify API.
+ * @requires request
+ * @requires bluebird
+ */
+'use strict';
+const request = require('request');
+const bluebird = require('bluebird');
```

Comment:
The overhead of using native promises is minimal in this case, it's not something to worry about IMO.